src/main/io/ledstrip.c

```diff
@@ -660,32 +650,62 @@ void applyLedWarningLayer(uint8_t updateNow)
     }
 }
 
+void applyLedGpsLayer(bool updateNow)
+{
+    static uint8_t gpsFlashCounter = 0;
+    static uint8_t gpsPauseCounter = 0;
+    const uint8_t blinkPauseLength = 4;
+
+    const hsvColor_t *gpsColor = &hsv_black;
+
+    if (GPS_numSat == 0 || !sensors(SENSOR_GPS)) {
+        gpsColor = &hsv_red;
+        gpsFlashCounter = gpsPauseCounter = 0; // reset counters
+    } else {
+        if (gpsPauseCounter == 0 && (gpsFlashCounter & 1) == 0) {
+            gpsColor = STATE(GPS_FIX) ? &hsv_green : &hsv_orange;
+        }
+    }
+
+    for (int i = 0; i < ledCount; ++i) {
+        const ledConfig_t *ledConfig = ledConfigs(i);
+
+        if (ledConfig->flags & LED_FUNCTION_GPS) {
+            setLedHsv(i, gpsColor);
+        }
+    }
+
+    if (updateNow) {
+        gpsFlashCounter++;
```

Comment:
`gpsFlashCunter` must be incremented only when `gpsPauseCounter` is zero (sorry, I overlooked that)
```
        if (gpsPauseCounter) {
            gpsPauseCounter--;
        } else {
           gpsFlashCounter ++;
           if (gpsFlashCounter >= GPS_numSat * 2) {
            gpsFlashCounter = 0;
            gpsPauseCounter = blinkPauseLength;
        }
  }